### PR TITLE
add lets split v3

### PR DESCRIPTION
**Pull request by ziptyze**
Initial commit for Let's Split v3

## Description

Addition of the Let's Split v3, a split ortholinear 4x12 keyboard with hotswap, in-switch RGB, and underglow. Powered by an rp2040 microcontroller

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ziptyze/lets_split_v3/config.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+#define SERIAL_USART_FULL_DUPLEX
+#define SERIAL_USART_TX_PIN GP16
+#define SERIAL_USART_RX_PIN GP17
+#define SERIAL_USART_PIN_SWAP
+
+#define RGBLED_NUM 126
+#define SPLIT_TRANSPORT_MIRROR
+#define RGBLIGHT_SPLIT
+#define RGB_MATRIX_LED_COUNT RGBLED_NUM
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/config.h:28]:
```suggestion
#define SPLIT_TRANSPORT_MIRROR
#define RGB_MATRIX_LED_COUNT 126
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:1]: This file is missing a license header.

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    _QWERTY,
+    _COLEMAK,
+    _DVORAK,
+    _LOWER,
+    _RAISE,
+    _ADJUST
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:13]:
```suggestion
  QWERTY = QK_USER,
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/rules.mk
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/rules.mk:1]: This file should be removed.

File: keyboards/ziptyze/lets_split_v3/readme.md
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/readme.md:1]: This is painfully copied from the [levinson readme](https://github.com/qmk/qmk_firmware/blob/develop/keyboards/keebio/levinson/readme.md). Please start over and follow the [template](https://github.com/qmk/qmk_firmware/blob/develop/data/templates/keyboard/readme.md).

File: keyboards/ziptyze/lets_split_v3/rules.mk [hunk 3]
```diff
@@ -0,0 +1,2 @@
+SPLIT_KEYBOARD = yes
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/rules.mk:1]:
```suggestion
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+)
+
+
+};
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:138]:
```suggestion
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_QWERTY);
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:143]:
```suggestion
        set_single_persistent_default_layer(_QWERTY);
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+      }
+      return false;
+      break;
+    case COLEMAK:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_COLEMAK);
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:149]:
```suggestion
        set_single_persistent_default_layer(_COLEMAK);
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+      }
+      return false;
+      break;
+    case DVORAK:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_DVORAK);
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:155]:
```suggestion
        set_single_persistent_default_layer(_DVORAK);
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+    case RAISE:
+      if (record->event.pressed) {
+        layer_on(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+      break;
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:178]: `TL_LOWR` and `TL_UPPR` should be used instead.
```suggestion
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+    case ADJUST:
+      if (record->event.pressed) {
+        layer_on(_ADJUST);
+      } else {
+        layer_off(_ADJUST);
+      }
+      return false;
+      break;
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:186]:
```suggestion
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+    _ADJUST
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  COLEMAK,
+  DVORAK,
+  LOWER,
+  RAISE,
+  ADJUST,
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:18]:
```suggestion
  DVORAK
```

File: keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,189 @@
...
+};
+
```
Comment (waffle87) [inline keyboards/ziptyze/lets_split_v3/keymaps/default/keymap.c:20]:
```suggestion

#define RAISE MO(_RAISE)

```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/ziptyze/lets_split_v3/config.h [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+#define SERIAL_USART_FULL_DUPLEX
+#define SERIAL_USART_TX_PIN GP16
+#define SERIAL_USART_RX_PIN GP17
+#define SERIAL_USART_PIN_SWAP
+
+#define SPLIT_TRANSPORT_MIRROR
+#define RGB_MATRIX_LED_COUNT 126
+#define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#define RGB_MATRIX_FRAMEBUFFER_EFFECTS
```
Comment (dunk2k) [inline keyboards/ziptyze/lets_split_v3/config.h:29]: KEYPRESSES and FRAMEBUFFER_EFFECTS are enabled when an `rgbmatrix.animation` is enabled 
```suggestion
#define RGB_MATRIX_LED_COUNT 126
#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```

File: keyboards/ziptyze/lets_split_v3/info.json [hunk 13]
```diff
@@ -0,0 +1,298 @@
...
+    "bootloader": "rp2040",
+    "processor": "RP2040",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "encoder": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": false,
```
Comment (dunk2k) [inline keyboards/ziptyze/lets_split_v3/info.json:15]: not enabled by default
```suggestion
        "extrakey": true,
        "mousekey": true,
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/ziptyze/lets_split_v3/info.json [hunk 14]
```diff
@@ -0,0 +1,298 @@
...
+        "rgb_matrix": true
+    },
+    "dynamic_keymap": {
+        "layer_count": 10
+    },
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x6900",
+        "vid": "0x6F75"
+    },
```
Comment (dunk2k) [inline keyboards/ziptyze/lets_split_v3/info.json:25]:
```suggestion
    },
    "split": {
        "enabled": true
    }
```

File: keyboards/ziptyze/lets_split_v3/info.json [hunk 15]
```diff
@@ -0,0 +1,298 @@
...
+            { "flags": 4, "matrix": [4, 0], "x": 121, "y": 8 },
+
+            { "flags": 2, "x": 113, "y": 12 },
+            { "flags": 2, "x": 113, "y": 4 },
+            { "flags": 2, "x": 117, "y": 1 },
+            { "flags": 2, "x": 140, "y": 1 },
+            { "flags": 2, "x": 149, "y": 1 },
+            { "flags": 2, "x": 159, "y": 1 },
+            { "flags": 2, "x": 169, "y": 1 }
+          ]
```
Comment (dunk2k) [inline keyboards/ziptyze/lets_split_v3/info.json:296]:
```suggestion
        ]
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/ziptyze/lets_split_v3/config.h [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+
+#pragma once
+
+#define SERIAL_USART_FULL_DUPLEX
+#define SERIAL_USART_TX_PIN GP16
+#define SERIAL_USART_RX_PIN GP17
+#define SERIAL_USART_PIN_SWAP
+
+#define SPLIT_TRANSPORT_MIRROR
+#define RGB_MATRIX_LED_COUNT 126
```
Comment (drashna) [inline keyboards/ziptyze/lets_split_v3/config.h:26]: This shouldn't be needed, as it should be generated automatically by the info.json processing.
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ziptyze/lets_split_v3/config.h [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+
+#pragma once
+
+#define SERIAL_USART_FULL_DUPLEX
+#define SERIAL_USART_TX_PIN GP16
+#define SERIAL_USART_RX_PIN GP17
+#define SERIAL_USART_PIN_SWAP
+
+#define SPLIT_TRANSPORT_MIRROR
+#define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
```
Comment (drashna) [inline keyboards/ziptyze/lets_split_v3/config.h:26]:
```suggestion
```

File: keyboards/ziptyze/lets_split_v3/info.json [hunk 18]
```diff
@@ -0,0 +1,299 @@
...
+    },
+    "dynamic_keymap": {
+        "layer_count": 10
+    },
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x6900",
+        "vid": "0x6F75"
+    },
+    "split": {
```
Comment (drashna) [inline keyboards/ziptyze/lets_split_v3/info.json:24]:
```suggestion
    "split": {
        "transport": {
            "sync": {
                "matrix_state": true
            }
        },
```

File: keyboards/ziptyze/lets_split_v3/info.json [hunk 19]
```diff
@@ -0,0 +1,299 @@
...
+                {"matrix": [7, 4], "x": 11, "y": 3},
+                {"matrix": [7, 5], "x": 12, "y": 3}
+            ]
+        }
+    },
+    "rgb_matrix": {
+        "driver": "ws2812",
+        "split_count": [63, 63],
+        "max_brightness": 200,
+        "animations": {
```
Comment (drashna) [inline keyboards/ziptyze/lets_split_v3/info.json:105]:
```suggestion
        "sleep": true,
        "animations": {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by waffle87 — APPROVED** [review]

File: keyboards/ziptyze/lets_split_v3/readme.md [hunk 20]
```diff
@@ -0,0 +1,26 @@
+# Let's Split v3
+
+![Let's Split v3](https://i.imgur.com/ksZ64dK.jpg)
```
Comment (fauxpark) [inline keyboards/ziptyze/lets_split_v3/readme.md:3]:
```suggestion
![Let's Split v3](https://i.imgur.com/ksZ64dKh.jpg)
```

File: keyboards/ziptyze/lets_split_v3/readme.md [hunk 21]
```diff
@@ -0,0 +1,26 @@
...
+
+Version 3 of the Let's Split keyboard, originally designed by Wootapatoot. This version features identical PCBs for both halves powered by an RP2040 controller. Each half features full hot swap capability as well as in-switch addressable LEDs and 39 1mm underglow LEDs.
+
+Keyboard Maintainer: [ziptyze](https://github.com/ziptyze)  
+Hardware Supported: Let's Split v3 PCB
+Hardware Availability: [1upkeyboards](https://1upkeyboards.com/shop/keyboard-kits/diy-40-kits/lets-split-v3/#choose-your-kit)
```
Comment (fauxpark) [inline keyboards/ziptyze/lets_split_v3/readme.md:9]:
```suggestion
* Keyboard Maintainer: [ziptyze](https://github.com/ziptyze)
* Hardware Supported: Let's Split v3 PCB
* Hardware Availability: [1upkeyboards](https://1upkeyboards.com/shop/keyboard-kits/diy-40-kits/lets-split-v3/#choose-your-kit)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**